modules/learn/pages/data/validity.adoc

```diff
@@ -0,0 +1,70 @@
+
+The meanings of these properties, and the ways they are achieved by Couchbase Server, are described below.
+Note that various references are made to https://jepsen.io[Jepsen] &#8212; a project dedicated to improving the safety of distributed databases and other systems, with reference to specific https://jepsen.io/consistency[Consistency Models].
+
+[#atomicity-definition-and-implementation]
+=== Atomicity
+
+_Atomicity_ guarantees that when multiple data-elements need to be modified such that only the successful modification of all justifies the modification of any, _either_ all the modifications do occur successfully; _or_ none of them occurs.
+
+Atomicity is innately supported by the xref:learn:data/document-data-model.adoc[Couchbase Data Model]; which is naturally expressive of hierarchical data, and therefore typically only requires a single document for data that, under a relational model, would require multiple tables; and would therefore maintain data-elements discretely.
+On Couchbase Server, changes made to a single document are all written simultaneously, thereby guaranteeing atomicity of all data-elements within the document for every write.
+
+Additionally, Couchbase Server supports the atomic inserting, updating, and deleting of data-elements across multiple documents, by means of xref:learn:data/transactions.adoc[Distributed ACID Transactions].
+
+[#consistency-definition-and-implementation]
+=== Consistency
+
+_Consistency_ requires that modifications to data occur only in accordance with required equivalencies and invariants.
+For example, multiple copies of a single data-element must all be updated identically, when an update to any one occurs.
+
+Couchbase Server provides xref:learn:clusters-and-availability/intra-cluster-replication.adoc[Intra-Cluster Replication]; which ensures data-availability with up to three replicas, deployed across the cluster.
+The _Database Change Protocol_ (DCP) keeps active and replica data-elements both consistent and up-to-date with _eventual consistency_.
+
+Additionally, Couchbase Server provides xref:learn:clusters-and-availability/xdcr-overview.adoc[Cross Data Center Replication (XDCR)]; which uses eventual consistency to replicate data-modifications across different clusters, potentially located in different data centers.
+
+The Couchbase xref:learn:services-and-indexes/services/index-service.adoc[Index Service] provides high-performance data-access; and as such, maintains fast-access copies of specified data.
+Such copies are updated with eventual consistency, across the cluster.
+A xref:n1ql:n1ql-rest-api/index.adoc#table_xmr_grl_lt[Read Your Own Writes] mechanism is available, to allow queries to read modified data from an index at the earliest point at which consistency is achieved.
+
+When data-elements must be updated consistently across multiple documents, xref:learn:data/transactions.adoc[Distributed ACID Transactions] can be used.
```

Comment:
I'd never seen a word reused for so many different purposes, until I came to work on a database and discovered "consistency"...

There's a lot of confusing elements in this section, I feel, as it's trying to define multiple sorts of consistency.  I don't think it's your fault, I think it's that we haven't defined super-well yet what consistency means for our transactions - in RMDBS, it traditionally referred to preserving things like foreign key constraints: things that don't apply to us.  Let me run this by Shivani.   

I think we should keep consistency strictly related to transactions here though.  This section is defining consistency for XDCR and indexing.

"When data-elements must be updated consistently across multiple documents," -> this is atomicity rather than consistency.

Reply:
GSI, FTS indexes, and XDCR are each referred to in the original doc as representative of 'eventual consistency', so I put them here. I assumed that on that basis, DCP was a valid contender as well. 

Other than referring to indexes and RYOW, the original doc says: "Synchronous Replication, available from Couchbase Data Platform 6.5 onwards, ensures documents are synchronously replicated to a majority of replicas."  So that sounds like a duplication of durability placed underneath consistency, and I guess doesn't really specify transactions. So perhaps I'm not sure what sure should go here, transaction-wise.